icu4c/source/samples/ucnv/convsamp.cpp

```diff
@@ -342,7 +342,7 @@ UErrorCode convsample_05()
         
         ucnv_toUnicode(conv, &target, targetLimit, 
                        &source, sourceLimit, NULL,
-                       feof(f)?TRUE:FALSE,         /* pass 'flush' when eof */
+                       feof(f)?true:false,         /* pass 'flush' when eof */
```

Comment:
Could this be the same as Daniel mentioned before? removing the condition? Same for cases below?

Reply:
Hmm, good question...

The documentation for the function `feof` only says that it returns a "nonzero value":
> The feof function returns a nonzero value if a read operation has attempted to read past the end of the file; it returns 0 otherwise.

However, the function `ucnv_toUnicode` takes a `UBool`, which is a `typedef` for `int8_t`. This means that it **only** wants a value of `1` for things that are "true". So because of this I think we need to keep the condition here, as we need to possibly handle cases where `feof` returns back something like 2.
In other words, we need it to work like this: ` feof returns 2 --> true --> 1`.
So I think this needs to stay as-is.

Reply:
Ah that is a good point :D Thanks!